x/oracle/keeper/keeper.go

```diff
@@ -92,40 +93,34 @@ func (k Keeper) ProcessClaim(ctx sdk.Context, id string, validator sdk.AccAddres
 			return types.Status{}, types.ErrDuplicateMessage(k.Codespace())
 		}
 		prophecy.AddClaim(validator, claim)
-		//	//check if claimthreshold is passed
-		//	//if does
-		//		//check enough claims match and are valid
-		//		//update prophecy to be successful
-		//		//trigger minting
-		//		//save finalized prophecy to db
-		//		//return
-		//	//if doesnt
-		//		//check if failure threshold passed
-		//		//if does, fail and update and return
-		//		//else, save updated prophecy to db and return
-		return types.Status{}, sdk.ErrInternal("Not yet implemented")
 	} else {
 		if err.Code() != types.CodeProphecyNotFound {
 			return types.Status{}, err
 		}
-		newProphecy := types.NewProphecy(id)
-		newProphecy.AddClaim(validator, claim)
-		err := k.saveProphecy(ctx, newProphecy)
-		if err != nil {
-			return types.Status{}, err
-		}
-		//return result
-		return prophecy.Status, nil
+		prophecy = types.NewProphecy(id)
+		prophecy.AddClaim(validator, claim)
 	}
+	prophecy = k.processCompletion(ctx, prophecy)
+	err = k.saveProphecy(ctx, prophecy)
+	if err != nil {
+		return types.Status{}, err
+	}
+	return prophecy.Status, nil
 }
 
-func (k Keeper) getPowerThreshold() int {
-	minimumPower := float64(getTotalPower()) * k.consensusNeeded
-	return int(math.Ceil(minimumPower))
-
-}
-
-func getTotalPower() int {
-	//TODO: Get from Tendermint/last block/staking module?
-	return 10
+func (k Keeper) processCompletion(ctx sdk.Context, prophecy types.Prophecy) types.Prophecy {
```

Comment:
Thoughts on extracting some of these logic operations out to a library for clarity?

Reply:
I think it's fine for now as it's just this 1 function